### PR TITLE
a few updates for the console menu /etc/rc.initial

### PR DESCRIPTION
**Pull request by luckman212**
- [x] Redmine Issue: https://redmine.pfsense.org/issues/13258
- [x] Ready for review

### changes
- add full pathnames to all binaries (before some were and some weren't)
- less forking for process checking, instead of ps | grep, use pgrep
- clean up spaces vs tabs
- remove the text "cheap menu"
- fixup hidden option 100 (links browser)

**n.b.** the `patch` command is failing to automatically patch this via system patches, I think because the open source version differs so much from the pfS+ version that has hardware-specific elements. So for testing, I had to manually patch my 6100 to test it. Not sure if there is a way to get it to cooperate with System Patches, maybe breaking it up into a set of smaller commits, but hopefully these changes are small enough that it can be reviewed without all that extra fuss.

### CODE REVIEW

**Comment by luckman212** [comment]
**n.b.** the dashboard may not actually load with the `links` browser ... but I was able to load other pages e.g. google. So, might still be useful to leave it in.

image

image

File: src/etc/rc.initial [hunk 1]
```diff
@@ -151,18 +148,21 @@ case ${opmode} in
...
-		links "http://localhost"
+	protocol=$(/usr/local/sbin/read_xml_tag.sh string system/webgui/protocol)
+	port=$(/usr/local/sbin/read_xml_tag.sh string system/webgui/port)
+	if [ -z "$port" ]; then
+		case $protocol in
+			https) port=443;;
+			*)     port=80;;
+		esac
 	fi
+	links "${protocol}://localhost:${port}"
```
Comment (jim-p) [inline src/etc/rc.initial:159]: `protocol` could be empty, so it needs an extra test to drop a default value in there if it is (`http`).

**Review by jim-p — CHANGES_REQUESTED** [review]

**Comment by luckman212** [comment]
Didn't know protocol could be empty. Ok I will rework to account for that.

**Comment by luckman212** [comment]
Should be good now @jim-p

**Comment by luckman212** [comment]
Couple more very minor touchups

- removed `11 | 111)` -- why was that there? to catch typos?
- consistency with single vs double quotes
- removed a couple of stray `;`s

File: src/etc/rc.initial (hunk 1 above)
Comment (luckman212) [reply src/etc/rc.initial:160]: Didn't know protocol could be empty. Ok I will rework to account for that.

**Review by jim-p — APPROVED** [review]

**Pull request merged**